helix/appendix/alternate-visual-studio-structures.rst

```diff
@@ -0,0 +1,231 @@
+is based on. Based on these principles, Visual Studio projects form a "physical" boundary
+and unit of change for modules in Sitecore Helix.
+
+Martin however makes an assumption that modules (packages) are released / deployed
+independently. Due to the versioning and testing complexity this would put on the typical Sitecore
+solution, Sitecore Helix actually `discourages partial deploys of modules </devops/deployment/strategy>`__,
+meaning that the unit of release is the entirety of the application. If we are not deploying
+and versioning these assemblies independently, it is a valid question as to whether
+the modules really merit their own Visual Studio projects.
+
+The modular architecture practices of Sitecore Helix though are meant to improve solution
+*maintainability*. The convention of using a separate web application
+project for each module (which deploys to Sitecore CM/CD) means that all the code and
+configuration related to a module are located in one place. Combine that with the module
+filesystem and solution folder structure, and you also have the items, tests, and other related
+code for that module. Practically, it's very easy for a future developer to find/discover
+the code related to a desired system change. It's also easier to track and manage the dependencies
+to and from that module.
+
+It is true that `FxCop rules <https://www.teamdevelopmentforsitecore.com/Blog/sitecore-helix-fxcop-rules>`__
+can be used to enforce Sitecore Helix dependency rules without separate projects, effectively giving you a
+"logical" boundary between modules to help enforce dependencies. However the loss of physical *Common Closure*
+and the potential maintenance impact should definitely be considered in taking this approach.
+
+
+How to structure when consolidating
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+How you consolidate the projects in your solution will ultimately depend on
+the issue(s) you are attempting to address, and your view on the purpose and role
+of Visual Studio projects.
+
+* You may group Features into combined projects based on the concept of
+  `Feature Groups </principles/architecture-principles/modules.html#groups>`__. This will
+  reduce the number of projects but care must be taken to avoid dependencies between
+  Features in a group.
+* You may place an entire Layer into its own project as a "logical" grouping of modules.
+  This benefits the Feature Layer most, as it should contain far more modules than other layers.
+  Consolidating the Project Layer likely provides little benefit due to the number of modules, and
+  may create confusion around the composition of each Project Module. You should
+  take care when consolidating the Foundation Layer as well, as the allowance for
+  Foundation-to-Foundation references means that these dependencies need to be closely
+  monitored and managed.
+* You can place all the code for a single deployment target (e.g. main Sitecore website CM/CD code)
```

Comment:
That is...gross ;)

Reply:
![](https://media.giphy.com/media/pVAMI8QYM42n6/giphy.gif)